api/admin/controller/__init__.py

```diff
@@ -1563,7 +1578,7 @@ def _mirror_integration_setting(self):
         }
         for integration in integrations:
             mirror_integration_setting['options'].append(
-                dict(key=integration.id, label=integration.name)
+                dict(key=str(integration.id), label=integration.name)
```

Comment:
This looks like you're setting up a ConfigurationSetting where `key` is "{type}_mirror_integration_id" and `value` is the database ID of another ExternalIntegration. But this shouldn't be a ConfigurationSetting at all -- it's an ExternalIntegration link where `purpose` is "{type}_mirror".

Reply:
No, I guess this is handled specially earlier in this file. But earlier in the file you need to check `.endswith("mirror_integration_id")`.